applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp

```diff
@@ -200,7 +219,7 @@ void  AddCustomUtilitiesToPython(pybind11::module& m)
         .def("ProjectNodalVariableOnTangentPlane", &GeometryUtilities::ProjectNodalVariableOnTangentPlane)
         .def("ExtractBoundaryNodes", &GeometryUtilities::ExtractBoundaryNodes)
         .def("ExtractEdgeNodes", &GeometryUtilities::ExtractEdgeNodes)
-        .def("ComputeDistancesToBoundingModelPart", &GeometryUtilities::ComputeDistancesToBoundingModelPart)
+        .def("ComputeDistancesToBoundingModelPart", _ComputeDistancesToBoundingModelPart)
```

Comment:
Have you tried to directly expose the fct?
It should convert list to vector automatically (remember to include pybind/stl.h)

Reply:
Agree. I did something similar a few days ago.

Reply:
I tried, does not work. From pybind docs:

> However, a fundamental limitation of this approach is that internal conversions between Python and C++ types involve a copy operation that prevents pass-by-reference semantics. 

Since here i need to pass by reference i would need to make the vector "opaque", that has some other implications and as far as i remember we should not use that in Kratos?

Reply:
Since i have two return vectors, i can also not return by value...

Reply:
@rubenzorrilla maybe you can point mw to your change from a few days ago, maybe i am missing something.

Reply:
Sorry @armingeiser, I didn't note the reference. Sorry for the noise.

Reply:
@rubenzorrilla all good, always appreciate your comments!